metpy/calc/basic.py

```diff
@@ -114,8 +123,8 @@ def wind_components(speed, wdir):
 
     """
     wdir = _check_radians(wdir, max_radians=4 * np.pi)
-    u = -speed * np.sin(wdir)
-    v = -speed * np.cos(wdir)
+    u = np.nan_to_num(-speed.magnitude * np.sin(wdir)) * speed.units
```

Comment:
What's this change trying to do?

Reply:
This is trying to handle wind directions of NaN for calm winds. Looking at it again, I think I should just wrap the direction, not the full product, which should also solve the quantity/series issue. But, if we just have the calm->0, north->360 like mentioned above, we wouldn't have to worry about it?